lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -433,6 +435,8 @@ sub _show {
             show_autoinst_log  => $job->should_show_autoinst_log,
             show_investigation => $job->should_show_investigation,
             show_live_tab      => $job->state ne DONE,
+            testgiturl         => $test_giturl,
+            needlegiturl       => $needle_giturl,
```

Comment:
Since this is only required for contents *within* the investigation tab it would make more sense to determine these variables only when rendering this tab.

Reply:
it makes sense. but i thought that it really not destructing and it can be useful in other cases as well.

Reply:
I don't know what you mean by that. If I understand Martchus correctly this should say:

```suggestion
            testgiturl         => gitrepodir(distri => $job->DISTRI),
            needlegiturl       => gitrepodir(distri => $job->DISTRI, needles => 1),
```

Reply:
I must still say it would make more sense to render these URLs on the page where they are actually needed (and this is the tab and not the overall page).